infra/src/main/kotlin/fi/fta/geoviite/infra/tracklayout/DraftableObjectDao.kt

```diff
@@ -98,6 +99,21 @@ abstract class DraftableDaoBase<T : Draftable<T>>(
             if (found.none { f -> f.officialId == id }) throw NoSuchEntityException(table.name, id)
         } }
     }
+
+    override fun fetchAllPublicationVersions(): List<PublicationVersion<T>> {
```

Comment:
Nämä 2 funkkaria (tämä ja ylempi) on niin lähellä toisiaan että voisi ehkä yhdistää, esim niin että tuo listaparametri on nullable ja jos se on null, palautetaan kaikki. Tyhjät listat ei leiki tuon SQL:n kanssa kivasti mutta luulisin että null ei ole ongelma. Jos ids-parametriin laittaa param mappiin arvon null, luulisin että se näkyy sql:ssä tyyliin `id in (null)` joka ei ole syntaksivirhe vaan ihan vain false. Eli silloin tuon checkin voisi vaihtaa muotoon `where (:ids is null or id in (:ids)) and draft = true`

Reply:
Joo, hyvä ehdotus. Jätin nämä alunperin tarkoituksella erillisiksi funkkareiksi koska en tykkää nullable-listaparametreista, mutta tässä voisi olla ihan validi käyttökohde yhdistämisellekin.